lib/broadway/rate_limiter.ex

```diff
@@ -0,0 +1,68 @@
+defmodule Broadway.RateLimiter do
+  @moduledoc false
+
+  use GenServer
+  @row_name :rate_limit_counter
+
+  def start_link(opts) do
+    case Keyword.fetch!(opts, :rate_limiting) do
+      # If we don't have rate limiting options, we don't even need to start this rate
+      # limiter process.
+      nil ->
+        :ignore
+
+      rate_limiting_opts ->
+        name = Keyword.fetch!(opts, :name)
+        genserver_opts = [name: Module.concat(name, __MODULE__)]
```

Comment:
We should probably concat only `RateLimiter`. Also, use the same name for the process and the ETS table, makes it easier to debug live systems.